tests/simulation/simul_noisify_2.py

```diff
@@ -0,0 +1,138 @@
+"""
+This is file contains a sample main function.
+You can build your own, to simulate and test your data.
+
+It is not a TEST file, it is a sample file to run the algorithm.
+"""
+
+import pdb
+from typing import OrderedDict
+from xmlrpc.client import boolean
+from src.dedupe_gazetteer import *
+from src.dedupe_gazetteer_utils import *
+
+
+import logging
+from datetime import datetime
+import optparse
+import _io, os
+import unittest
+import dedupe
+import itertools
+from collections import defaultdict
+from settings import *
+
+
+
+def train(variables):
+    training_element = TrainingElement(op_training_file, variables)
+    """
+    Skipt training with you have already trained otherwhise it will delete your trained and settings file,
+    and train again.
+    """
+    tp = TrainingProcess(ip_canonical_file, op_settings_file, op_training_file, training_element)
+    print(f"Number of records from canonical data (pesquisadores unicos): {len(tp.canonical_d)}")
```

Comment:
This line is not more necessary. The print of 'canonical_d' size is in the 'training' method of the 'TrainingProcess' class.